File: fund-tracker/frontend/src/types/fund.ts
export interface Holding {
  holding_name: string;
  weight: number | null;
  holding_type: string | null;
  as_of_date: string | null;
}

export interface FundSummary {
  id: number;
  ticker: string | null;
  isin: string | null;
  internal_code: string | null;
  name: string;
  asset_class: string | null;
  style: string | null;
  currency: string;
  is_demo: boolean;
  latest_nav: number | null;
  latest_date: string | null;
}

export interface FundDetail extends FundSummary {
  objective: string | null;
  holdings: Holding[];
  created_at: string | null;
  updated_at: string | null;
}

File: fund-tracker/backend/app/schemas/fund.py
from pydantic import BaseModel
from datetime import date, datetime
from typing import Optional


class HoldingSchema(BaseModel):
    holding_name: str
    weight: Optional[float] = None
    holding_type: Optional[str] = None
    as_of_date: Optional[date] = None

    model_config = {"from_attributes": True}


class FundSummarySchema(BaseModel):
    id: int
    ticker: Optional[str] = None
    isin: Optional[str] = None
    internal_code: Optional[str] = None
    name: str
    asset_class: Optional[str] = None
    style: Optional[str] = None
    currency: str
    is_demo: bool
    latest_nav: Optional[float] = None
    latest_date: Optional[date] = None

    model_config = {"from_attributes": True}


class FundDetailSchema(FundSummarySchema):
    objective: Optional[str] = None
    holdings: list[HoldingSchema] = []
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None


class FundSearchRequest(BaseModel):
    query: str


class FundSearchResponse(BaseModel):
    fund: FundSummarySchema
    created: bool
